trin-core/src/utils/bytes.rs

```diff
@@ -27,6 +29,21 @@ pub fn random_32byte_array(leading_bit_zeros: u8) -> [u8; 32] {
     bytes
 }
 
+/// Encode hex with 0x prefix
+pub fn hex_encode<T: AsRef<[u8]>>(data: T) -> String {
+    format!("0x{}", hex::encode(data))
+}
+/// Decode hex with 0x prefix
+pub fn hex_decode(data: &str) -> anyhow::Result<Vec<u8>> {
+    let first_two = &data[..2];
+    match first_two {
+        "0x" => hex::decode(&data[2..]).map_err(|e| e.into()),
```

Comment:
The other option I considered for this was:
```suggestion
        "0x" => Ok(hex::decode(&data[2..])?),
```

It's a little shorter, but a little less obvious to my eyes. (Though that may still just be my newness). So it was my second favorite option. But open to other opinions/suggestions!